File: C/enzoniko-cs50-problems-2020-x-readability/readability.c

#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int letterscount = 0;
    int wordcount = 1;
    int sentencecount = 0;

//count words
    for (int i = 0; i < strlen(text); i++)
    {
       if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
       {
           letterscount++;
       }
       else if (text[i] == ' ')
       {
           wordcount++;
       }
       else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
       {
           sentencecount++;
       }
    }
//    printf("letters: %i; words: %i; sentences: %i\n", letterscount, wordcount, sentencecount);

    float grade = 0.0588 * (100 * (float) letterscount / (float) wordcount) - 0.296 * (100 * (float) sentencecount / (float) wordcount) - 15.8;
    if (grade < 16 && grade >= 0)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

}

File: Python/enzoniko-cs50-problems-2020-x-sentimental-readability/readability.py
from cs50 import get_string


text = get_string("Text: ")
letterscount = 0
wordcount = 1
sentencecount = 0
for i in range(len(text)):
    if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
        letterscount += 1
    elif (text[i] == " "):
        wordcount += 1
    elif text[i] in [".", "!", "?"]:
        sentencecount += 1
letterscount = 1.0 * letterscount
wordcount = 1.0 * wordcount
sentencecount = 1.0 * sentencecount
g = 0.0588 * ( 100 * (letterscount / wordcount)) - 0.296 * ( 100 * (sentencecount / wordcount)) - 15.8

grade = (round(g))
if (g < 16 and g >= 0):
    print("Grade: ", grade)
elif (g >= 16):
    print("Grade 16+")
else:
    print("Before Grade 1")

